contracts/protocol/lendingpool/Looping.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.6.12;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ILendingPool} from '../../interfaces/ILendingPool.sol';
+import {ILendingPoolAddressesProvider} from '../../interfaces/ILendingPoolAddressesProvider.sol';
+import "../../misc/interfaces/IWETH.sol";
+
+contract Looping {
```

Comment:
We can move the contract to `misc` folder. Similar to `WETHGateway`, `Looping` extends the functionalities of core lending pool contracts.